CRM/Civicase/Form/Report/BaseExtendedReport.php

```diff
@@ -536,25 +769,38 @@ public function addColumnAggregateSelect($fieldName, $dbAlias, array $spec) {
       return;
     }
     $spec['dbAlias'] = $dbAlias;
-    $options = $this->getCustomFieldOptions($spec);
-
-    if (!empty($this->_params[$fieldName . '_value']) && CRM_Utils_Array::value($fieldName . '_op', $this->_params) == 'in') {
-      $options['values'] = array_intersect_key($options, array_flip($this->_params[$fieldName . '_value']));
+    if (CRM_Utils_Array::value('data_type', $spec) === 'ContactReference') {
```

Comment:
Would have preferred we still keep this function simple and instead create a new function that returns options based on the type. e.g

```php
  private function getFieldOptions($spec, $fieldName, $dbAlias) {
    if (CRM_Utils_Array::value('data_type', $spec) === 'ContactReference') {
      $this->_aggregatesIncludeNULL = FALSE;
      $usedIds = array_keys($this->getUsedOptions($dbAlias, $fieldName));
      if (empty($usedIds)) {
        return;
      }

      return $this->getContactColumnOptions($spec, $usedIds);
    }

    $options = $this->getCustomFieldOptions($spec);
return $options = array_intersect_key($options, $this->getUsedOptions($dbAlias, $fieldName));

  }

``` 
Then in the `addColumnAggregateSelect` function

```php
 $options = $this->getFieldOptions($spec, $fieldName, $dbAlias);

```

It doesn't have to be exactly like this, the idea is to keep the `addColumnAggregateSelect` simple